src/crawlers/playwright_crawler.js

```diff
@@ -299,6 +299,42 @@ class PlaywrightCrawler extends BrowserCrawler {
         this.launchContext = launchContext;
     }
 
+    /**
+     * Returns context options that will be used when a page needs to be created.
+     * Required for proxy per page.
+     *
+     * @param {BrowserCrawlingContext & CrawlingContext} crawlingContext
+     * @ignore
+     * @protected
+     * @internal
+     */
+    _getNewPageOptions(crawlingContext) {
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
+            const { url } = this.proxyConfiguration.newProxyInfo(session && session.id);
+
+            const parsed = new URL(url);
+
+            pageOptions.proxy = {
+                server: parsed.origin,
+                username: parsed.username,
+                password: parsed.password,
+            };
+
+            // Disable SSL verification for MITM proxies
+            if (this.proxyConfiguration.isManInTheMiddle) {
+                /**
+                 * @see https://playwright.dev/docs/api/class-browser/#browser-new-context
+                 */
+                pageOptions.ignoreHTTPSErrors = true;
+            }
```

Comment:
This part could be in `BrowserCrawler` no?

Reply:
I was thinking about this as well, but since there's different proxy properties for both puppeteer and playwright I decided to treat this like two separate entities - even if it's duplicated, it makes the code clear. Making a separate function just for shared properties will imo just scatter the code.